contracts/templates/TemplateLauncher.sol

```diff
@@ -181,4 +189,16 @@ contract TemplateLauncher is CloneFactory {
     function getTemplateId(address _template) public view returns (uint256) {
         return templateToId[_template];
     }
+
+    /// @dev verifies that a template address implements the AquaTemplate standard
+    /// @param _template the template contract address
+    function isTemplateCompliant(address _template) private returns (bool) {
+        try ITemplate(_template).templateName() {
+            return true;
+        } catch (
+            bytes memory /*lowLevelData*/
+        ) {
+            return false;
+        }
+    }
 }
```

Comment:
Is this bulletproof? Also, what are the implications of this on gas efficiency?

Reply:
Hey Adam – we could add this functionality to make a technical check – even though this won't be bulletproof. Our approach until now was that in theory also broken templates could be added, and the framework will be fine with it, as there would be a techical/governance process to check & verify the templates. Only double-checked templates would get the verified status.

If we want to implement it on contract level, we could move the logic in a modified instead a function.

Reply:
When it comes to gas efficiency increase is completely acceptable:
(Min / Max / Avg)
Pre:
![Screenshot from 2021-08-09 12-31-30](https://user-images.githubusercontent.com/26334170/128694777-86886121-1233-45e2-9616-ac029a80da0b.png)
Post;
![Screenshot from 2021-08-09 12-36-42](https://user-images.githubusercontent.com/26334170/128694800-6c25df2d-b1e2-47aa-a8c7-f7c71ac1af22.png)